orca-spring-batch/src/main/java/com/netflix/spinnaker/zombie/ZombiePipelineCleanupAgent.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.zombie;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import com.netflix.spinnaker.orca.RetryableTask;
+import com.netflix.spinnaker.orca.pipeline.model.Pipeline;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import com.netflix.spinnaker.orca.pipeline.model.Task;
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+import static com.netflix.spinnaker.orca.ExecutionStatus.CANCELED;
+import static com.netflix.spinnaker.orca.ExecutionStatus.RUNNING;
+import static java.time.Duration.ZERO;
+import static java.time.Duration.between;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static rx.Observable.just;
+import static rx.Observable.timer;
+
+/**
+ * This poller cancels any pipelines that appear to be still running but are
+ * more than {@link #TOLERANCE} overdue.
+ */
+@Component
```

Comment:
Where does this actually get scanned by spring? I don't see this package or class referenced via `Main` or somewhere similar.